src/transformers/training_args.py

```diff
@@ -1324,20 +1359,41 @@ def __post_init__(self):
             getattr(self.fsdp_config, "fsdp_min_num_params", 0), self.fsdp_min_num_params
         )
 
+        if self.fsdp_transformer_layer_cls_to_wrap is not None:
+            warnings.warn("using `--fsdp_transformer_layer_cls_to_wrap` is deprecated. Use fsdp_config instead ", FutureWarning)
+            self.fsdp_config["fsdp_transformer_layer_cls_to_wrap"] = getattr(self.fsdp_config, "fsdp_transformer_layer_cls_to_wrap", [])+[self.fsdp_transformer_layer_cls_to_wrap]
+
         if len(self.fsdp) == 0 and self.fsdp_config["fsdp_min_num_params"] > 0:
             warnings.warn("`--fsdp_min_num_params` is useful only when `--fsdp` is specified.")
 
-        if len(self.fsdp) == 0 and self.fsdp_transformer_layer_cls_to_wrap is not None:
+        if len(self.fsdp) == 0 and self.fsdp_config["fsdp_transformer_layer_cls_to_wrap"] is not None:
             warnings.warn("`--fsdp_transformer_layer_cls_to_wrap` is useful only when `--fsdp` is specified.")
 
         if (
             len(self.fsdp) > 0
             and self.fsdp_config["fsdp_min_num_params"] > 0
-            and self.fsdp_transformer_layer_cls_to_wrap is not None
+            and self.fsdp_config["fsdp_transformer_layer_cls_to_wrap"] is not None
         ):
             raise ValueError(
                 "`--fsdp_min_num_params` and `--fsdp_transformer_layer_cls_to_wrap` are mutually exclusive."
             )
+        self.fsdp_config["xla"] = getattr(self.fsdp_config, "xla", False)
+        self.fsdp_config["xla_fsdp_grad_ckpt"] = getattr(self.fsdp_config, "xla_fsdp_grad_ckpt", False)
+        if self.fsdp_config["xla"]:
+            if self.fsdp is not "":
+                # gather fsdp configuration parameters into a dictionary from specified json file
+                with io.open(self.fsdp_config["xla_fsdp_settings"], "r", encoding="utf-8") as f:
+                    self.xla_fsdp_config = json.load(f)
+                # apply appropriate string to torch.dtype conversions for parameters
+                if "compute_dtype" in self.xla_fsdp_config:
+                    self.xla_fsdp_config["compute_dtype"] = getattr(torch, self.xla_fsdp_config["compute_dtype"])
+                if "buffer_dtype" in self.xla_fsdp_config:
+                    self.xla_fsdp_config["buffer_dtype"] = getattr(torch, self.xla_fsdp_config["buffer_dtype"])
```

Comment:
Why not use `self.fsdp_config` here?

Reply:
Sorry, I'm not sure I understand the question. Are you asking why have a separate dictionary of XLA FSDP settings, or why have the flag `xla_fsdp_settings` at all? If you're asking about the latter, this is what I mentioned in my comment to @pacman100 above: the [Fairscale FSDP arguments](https://github.com/facebookresearch/fairscale/blob/v0.4.13/fairscale/nn/data_parallel/fully_sharded_data_parallel.py#L170-L291) are different enough from the [XLA FSDP arguments](https://github.com/pytorch/xla/blob/master/torch_xla/distributed/fsdp/xla_fully_sharded_data_parallel.py#L122-L240) that one cannot reuse the Fairscale FSDP settings for XLA FSDP, and adding a separate flag for each non-redundant XLA FSDP parameter seems like it might bloat the fsdp_config file with too many XLA specific arguments. 

If you're asking about the former, I eventually pass `xla_fsdp_config` as `**kwargs` to the XLA FSDP wrapper, so (as far as I know) loading the JSON file storing the XLA configuration options into the `fsdp_config` dictionary would make this no longer work. This could be a gap in my Python though - perhaps the FSDP wrapper would ignore all unused arguments if I passed in `fsdp_config` instead?

Apologies if I've understood your question - if so, please let me know.

Reply:
I'm not too sure what Faiscale has to do with FSDP here, we use the PyTorch FSDP. Fairscale is the old implementation hidden behind the `--sharded_ddp` argument but we're not maintaining it anymore since it was never really used.

Reply:
Sorry for the confusion - this is an error on my part. I was under the impression that Fairscale FSDP and PyTorch FSDP referred to the same implementation, but I see that this is not the case. That being said, I think my previous comments still apply to PyTorch FSDP, which is what I had intended to refer to: looking at the [PyTorch FSDP documentation](https://pytorch.org/docs/stable/fsdp.html), the parameters for PyTorch FSDP wrapping do not overlap much with the parameters for XLA FSDP wrapping.